src/pages/scan.tsx

```diff
@@ -1,25 +1,45 @@
-import React, { useState } from 'react';
+import React, { use, useEffect, useState } from 'react';
 import { QrReader } from 'react-qr-reader';
 import { motion } from 'framer-motion';
 import { XMarkIcon } from '@heroicons/react/24/solid';
 import Link from 'next/link';
 import { useRouter } from 'next/router';
+import { useToast } from '@/components/Toast';
+import { httpPost } from '@/utils/axios';
+import { useAuth } from '@/context/AuthContext';
 
 function Scan() {
+    const { isAuthenticated, isReady } = useAuth();
     const [data, setData] = useState('');
     const [showModal, setShowModal] = useState(false);
-
-    const handleScanResult = (result: any, error: any) => {
-        if (result) {
-            setData(result.text);
+    const toast = useToast();
+    const router = useRouter();
+    const checkIn = async (token: any) => {
```

Comment:
after scan you should setData("") so, It will not be looped.

Reply:
@boomchanotai done :D